app/adapters/alma_adapter.rb

```diff
@@ -46,6 +46,95 @@ def get_availability_many(ids:)
     availability
   end
 
+  def get_availability_holding(id:, holding_id:, query: nil)
+    # Fetch the bib record and get the information for the individual holding
+    bibs = Alma::Bib.find(Array.wrap(id), expand: ["p_avail", "e_avail", "d_avail", "requests"].join(",")).each
+    return nil if bibs.count == 0
+
+    bib_status = AvailabilityStatus.from_bib(bib: bibs.first)
+    marc_holding = bib_status.holding(holding_id: holding_id) || {}
+
+    # Fetch the items for the holding...
+    holding_items = bib_status.holding_item_data(holding_id: holding_id, query: query)
+
+    # This is to protect in case the bib_id and the holding_id are valid but not related
+    # i.e. the holding_id is for a different bib_id than the one we received.
+    #
+    # In this case Alma will return the correct `total_count` of items for the holding_id
+    # (since the holding_id exists) but it will not return the items for the holding
+    # because the holding_id belongs to a different bib_id.
+    return nil if holding_items[:items].count == 0 && holding_items[:total_count] > 0
```

Comment:
This has now been removed since we are using the Alma gem to do the call and it does not handle this kind of granularity. The user will get zero records if there is a missmatch between the `bib_id` and the `holding_id`